app/javascript/components/upload/UploadWizard.js

```diff
@@ -54,12 +54,24 @@ const STEPS = [
 const MAIN_STEPS = STEPS.slice(0, 4)
 const SUPPLEMENTAL_STEPS = STEPS.slice(4)
 
+const defaultStudyState = {
+  accession: null,
+  url_safe_name: null
+}
+
+// context to pass through UploadWizard
+export const StudyContext = React.createContext(defaultStudyState)
 
 /** shows the upload wizard */
 export function RawUploadWizard({ studyAccession, name }) {
   const [serverState, setServerState] = useState(null)
   const [formState, setFormState] = useState(null)
 
+  // study attributes to pass to the StudyContext later for use throughout the RawUploadWizard component, if needed
+  // use complete study object, rather than defaultStudyState so that any updates to study.rb will be reflected in
+  // this context
+  const studyObj = serverState?.study
```

Comment:
Yeah, originally I thought we'd want to limit the properties we pass to the context to just name and accession, but you're right that using the full study object as the context will cause a re-render of all the components that use the study context,  when attributes change.  And so if we ever want to have, e.g. a 'usage' tab in the upload wizard with a real-time updating view count, this will let us do it

Reply:
On thinking about this more, this setup will cause the components that rely on the studyContext to get rerendered every time we poll the server for study state (since the serverState object itself will be a different instance, even though it will have all the same property values.  So maybe not optimal, but totally fine for now given that we poll the server only every 10s, and not that many components rely on this context